java-util/src/main/java/io/druid/java/util/common/parsers/AbstractFlatTextFormatParser.java

```diff
@@ -154,7 +181,8 @@ public void setFieldNames(final String header)
         setFieldNames(ParserUtils.generateFieldNames(values.size()));
       }
 
-      return Utils.zipMapPartial(fieldNames, Iterables.transform(values, valueFunction));
+      Map<String, String> fieldValueMap = Utils.zipMapPartial(fieldNames, values);
```

Comment:
I suggest to populate a HashMap in a simple loop, without `multiValueTransformer` functional object, but rather having it's functionality in a method in `AbstractFlatTextFormatParser` class.

Reply:
I don't exactly know the benifits for doing this, On the contrary `multiValueTransformer` looks more simply and make the AbstractClass clear for users.

Reply:
There are no benefits, in fact it's quite obscure (I don't think anybody can tell what does "zipMapPartial" does without going to it's implementation.)

Creating a functional object and storing it in a field of a class is an antipattern, versus moving the same logic in a method of that class. This method could be private, so it doesn't affect the class users in any way.